File: scalable-data-visualization-solution-advanced/backend/app/schemas/visualization.py
```python
from typing import Optional, Dict, Any
from datetime import datetime
import uuid

from pydantic import BaseModel, Field

class ChartType(str, Enum):
    BAR_CHART = "BarChart"
    LINE_CHART = "LineChart"
    PIE_CHART = "PieChart"
    AREA_CHART = "AreaChart"
    SCATTER_CHART = "ScatterChart"

class VisualizationBase(BaseModel):
    name: str = Field(..., min_length=1, max_length=100)
    description: Optional[str] = Field(None, max_length=500)
    dataset_id: int
    chart_type: ChartType
    chart_config: Dict[str, Any] = Field(..., description="Configuration specific to the chart type, e.g., axes, labels, colors.")

class VisualizationCreate(VisualizationBase):
    pass

class VisualizationUpdate(VisualizationBase):
    name: Optional[str] = None
    description: Optional[str] = None
    dataset_id: Optional[int] = None
    chart_type: Optional[ChartType] = None
    chart_config: Optional[Dict[str, Any]] = None

class VisualizationInDBBase(VisualizationBase):
    id: int
    uuid: uuid.UUID
    owner_id: int
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True

class Visualization(VisualizationInDBBase):
    pass

class VisualizationInDB(VisualizationInDBBase):
    pass
```

File: modern-data-visualization-platform-pro/frontend/src/types/visualization.ts
import { Dataset } from './dataset';
import { Dashboard } from './dashboard';

/**
 * Enum for supported chart types.
 */
export enum ChartType {
  Bar = 'bar',
  Line = 'line',
  Pie = 'pie',
  Table = 'table',
}

/**
 * Represents a visualization object.
 */
export interface Visualization {
  id: string;
  ownerId: string;
  name: string;
  description: string;
  type: ChartType;
  config: Record<string, any>; // JSON configuration specific to the chart type
  datasetId: string;
  dataset?: Dataset; // Optional, might be eager loaded for data processing
  dashboardId?: string | null;
  dashboard?: Dashboard; // Optional, might be eager loaded for context/ownership
  createdAt: string;
  updatedAt: string;
}

/**
 * Represents processed data ready for charting.
 * The structure depends on the chart library and type (e.g., Chart.js data object).
 */
export interface ChartData {
  labels: string[];
  datasets: Array<{
    label: string;
    data: number[];
    backgroundColor?: string[];
    borderColor?: string[];
    borderWidth?: number;
    hoverBackgroundColor?: string[];
    // Add other common chart.js dataset properties
  }>;
}

/**
 * Represents processed data for a table visualization.
 */
export interface TableData {
  columns: Array<{ field: string; headerName: string; width: number }>;
  rows: Record<string, any>[];
}

/**
 * Generic type for processed visualization data, can be ChartData or TableData.
 */
export type ProcessedVisualizationData = ChartData | TableData;